### ISSUE TITLE
Incremental file install not working if previous source re-used

### ISSUE THREAD

**Issue by dpmatthews** [metomi/rose#2438]
If I run the following
```
echo 1 > test1
echo 2 > test2
echo 3 > test3
mkdir app-run
cd app-run
printf "[command]\ndefault=\n[file:test]\nsource=$OLDPWD/test1" > rose-app.conf 
rose app-run -q
echo  "Installed contents: $(cat test)"
printf "[command]\ndefault=\n[file:test]\nsource=$OLDPWD/test2" > rose-app.conf 
rose app-run -q
echo  "Installed contents: $(cat test)"
printf "[command]\ndefault=\n[file:test]\nsource=$OLDPWD/test3" > rose-app.conf 
rose app-run -q
echo  "Installed contents: $(cat test)"
printf "[command]\ndefault=\n[file:test]\nsource=$OLDPWD/test1" > rose-app.conf 
rose app-run -q
echo  "Installed contents: $(cat test)"
```
I get
```
Installed contents: 1
Installed contents: 2
Installed contents: 3
Installed contents: 3
```
Changing the source back to a previous version is just ignored.

**Comment by wxtim** [comment]
I believe that this is not catered for in line 213 of `lib/python/rose/config_processors/fileinstall.py`.
```python
            source.is_out_of_date = (
                not prev_source or
                (not source.key and not source.paths) or
                prev_source.scheme != source.scheme or
                prev_source.loc_type != source.loc_type or
                prev_source.key != source.key or
                sorted(prev_source.paths) != sorted(source.paths))
```
I'm still investigating what `scheme`, `loc_type` and `key` are, but it doesn't look like a file with the same path as one in the DB is recognised as different. I presume that we'd like to store a checksum in the DB and check this?

**Comment by wxtim** [comment]
I've found and fixed this bug in Rose 1.x. I will look at whether this bug occurs in `cylc install` once https://github.com/cylc/cylc-flow/pull/4023 has been merged. The test created in #2440  should probably be ported to `cylc/cylc-rose`.

**Comment by oliver-sanders** [comment]
closed by #2440
